specifyweb/workbench/upload/upload_table.py

```diff
@@ -55,6 +56,28 @@ def to_json(self) -> Dict:
 
     def unparse(self) -> Dict:
         return { 'baseTableName': self.name, 'uploadable': self.to_json() }
+    
+class DeferredScopeUploadTable(NamedTuple):
```

Comment:
could make this class extend `UploadTable` rather than `NamedTuple` since most of the stuff can be inherited, right?

Reply:
That is the very first thing I tried. Unfortunately, it seemed to break any new attributes for the named tuple: You could not create an instance of the object with any newly declared attributes as keywords in the constructor.

I just have a basic understanding of Named Tuples to begin with, so I am sure there is a way to do it. 
I will do some further investigation.

Reply:
It seems that Named Tuples are not built/meant to be inherited from (see https://github.com/python/typing/issues/427 and this answer to a StackOverflow question https://stackoverflow.com/a/50369898).

Perhaps the easiest implementation is to create a new child class of NamedTuple whenever we need and manually add every function/attribute we need. 

However, this is not the 'cleanest' solution, and makes implementing future needed NamedTuples prone to bugs and 'harder' than it should be. 

There are some workarounds mentioned online, each with its own pros and cons. 
We could use dataclasses (described in the aforementioned StackOverflow question), the `@property` decorator (https://stackoverflow.com/a/55989641), or override each subclasses call to `__new__` (https://stackoverflow.com/a/42386174).

However, refactoring that bit of WorkBench code may require a decent amount of code changes and testing, so it is probably not in the scope of this pull request.

Reply:
Giving that this is a limitation of Mypy/python typing I have no problems with current solution given that workarounds are only adding more complexity

Though perhaps you could leave a brief comment in the code linking to relevant information so that if this get's fixed by python in the future, the code can be refactored accordingly